### PR TITLE
Build Mac PKG

### PR DESCRIPTION
**Pull request by sethAmazon**
# Description of the issue
There is no way to build pkg on github actions

# Description of changes
Add building of pkg on github actions

# Tests
Ran on my local fork.

### CODE REVIEW

File: integration/pkg/tools/create_pkg.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+fi
+
+mkdir /tmp/AmazonAgentScripts
+mv preinstall.sh /tmp/AmazonAgentScripts/preinstall
+mv postinstall.sh /tmp/AmazonAgentScripts/postinstall
+chmod +x /tmp/AmazonAgentScripts/preinstall
+chmod +x /tmp/AmazonAgentScripts/postinstall
+
+#We need to follow this directory structure for the Adhoc signing process
+#https://w.amazon.com/bin/view/Electric_Company/Website/API/adhoc-signing/#HStep3.Prepareyourpackagelocallyandupload
```
Comment (SaxyPandaBear) [inline integration/pkg/tools/create_pkg.sh:25]: Remove this.

File: integration/pkg/tools/postinstall.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+. /etc/rc.common
+sudo dscl . create /Users/cwagent
+sudo dscl . create /Users/cwagent RealName cwagent
+
+sudo dscl . create /Users/cwagent UniqueID $NextID
+# PrimaryGroupID of 20 to create a standard user
+sudo dscl . create /Users/cwagent PrimaryGroupID 20
+sudo dscl . create /Users/cwagent UserShell /usr/bin/false
+sudo dscl . create /Users/cwagent NFSHomeDirectory /Users/cwagent
```
Comment (SaxyPandaBear) [inline integration/pkg/tools/postinstall.sh:26]: Does this also need to create the aoc user?

Comment (sethAmazon) [reply integration/pkg/tools/postinstall.sh:26]: I'm not sure. I copied this code from the internal code. So I assume no but I have not tested.

Comment (sethAmazon) [reply integration/pkg/tools/postinstall.sh:26]: I will test this and report back.

Comment (sethAmazon) [reply integration/pkg/tools/postinstall.sh:26]: It seems aoc is not supported in darwin ctl. 
https://github.com/aws/amazon-cloudwatch-agent/blob/master/packaging/darwin/amazon-cloudwatch-agent-ctl#L36-L71 https://github.com/aws/amazon-cloudwatch-agent/blob/master/Makefile#L247-L259 not sure why this is not enabled

**Review by SaxyPandaBear — APPROVED** [review]

**Pull request merged**